src/core_api/db/node_database.py

```
@@ -0,0 +1,99 @@
+from typing import Dict, Optional, List, Any
+from pymongo import MongoClient
+from pymongo.errors import PyMongoError, DuplicateKeyError
+from ..models.node_model import Node
+from ..config import DATABASE_URL, DATABASE_CONNECT_TIMEOUT
+
+class NodeDatabase:
+    def __init__(self):
+        self.client = self._get_client()
+        self.db = self.client.get_database()
+        self.collection = self.db["nodes"]
+        self._check_indexes()
+
+    def _get_client(self) -> MongoClient[Any]:
+        try:
+            client: MongoClient[Any] = MongoClient(
+                DATABASE_URL,
+                connectTimeoutMS=DATABASE_CONNECT_TIMEOUT
+            )
+            client.admin.command('ping')
+            return client
+        except PyMongoError as node_db_error:
+            raise RuntimeError(f"MongoDB failed connecting: {node_db_error}")
+        # mongodb connection
+
+    def _check_indexes(self):
+        try:
+            self.collection.create_index("node_id", unique=True)
+        except PyMongoError as node_db_error:
+            raise RuntimeError(f"Index unresponsive: {node_db_error}")
+        # check indexing for node_id
+
+
+    def create_node(self, node_data: Dict[str, Any]) -> Node:
+        try:
+            node = Node(**node_data)
+            result = self.collection.insert_one(node.model_dump())
+            if not result.acknowledged:
+                raise RuntimeError("Failed to create node")
+            return node
+        except DuplicateKeyError:
+            raise ValueError(f"Node with ID {node_data.get('node_id')} already exists")
+        except PyMongoError as node_db_error:
+            raise RuntimeError(f"Creating operation failed: {node_db_error}")
+        # creating new node
+
+
+    def get_node(self, node_id: str) -> Optional[Node]:
```

Instead of returning None when no node was found raise and Exception